### PR TITLE
Update readme and environment.yml to make installation easier with conda+pip

### PR DESCRIPTION
**Pull request by erikmannerfelt**
Solves #119

### CODE REVIEW

**Comment by erikmannerfelt** [comment]
I also included some improvements to work with pypi while I was at it.

File: setup.py [hunk 1]
```diff
@@ -1,22 +1,45 @@
...
 setup(name='xdem',
       version=FULLVERSION,
-      description='',
-      url='',
+      description='Set of tools to manipulate Digital Elevation Models (DEMs) ',
+      long_description=LONG_DESCRIPTION,
+      long_description_content_type="text/markdown",
+      url='https://github.com/GlacioHack/xdem',
       author='The GlacioHack Team',
+      author_email="john@doe.com",
```
Comment (rhugonnet) [inline setup.py:34]: is john@doe.com really necessary here? :O

File: setup.py [hunk 2]
```diff
@@ -1,22 +1,45 @@
...
+import sys
 
 from setuptools import setup
+from setuptools.command.install import install
 
-FULLVERSION = '0.0.1'
+FULLVERSION = '0.0.2-2'
 VERSION = FULLVERSION
 
+class PostInstall(install):
```
Comment (rhugonnet) [inline setup.py:11]: can you add a short comment on what this does (for posterity)? :D

**Review by rhugonnet — APPROVED** [review]
Great :)
Nice spotting for the "glacio" conda environment name instead of "xdem"

File: setup.py (hunk 1 above)
Comment (erikmannerfelt) [reply setup.py]: Unfortunately yes... I don't remember how to reproduce the error, but I had to add an email address for everything to work!

Comment (erikmannerfelt) [reply setup.py]: Added a small comment for the future in a7734de

File: setup.py (hunk 2 above)
Comment (erikmannerfelt) [reply setup.py:11]: Added in a7734de

File: environment.yml [hunk 3]
```diff
@@ -22,5 +22,8 @@ dependencies:
     - richdem
     - scikit-gstat
     - pytransform3d
+    - https://github.com/GlacioHack/GeoUtils/tarball/master
```
Comment (adehecq) [inline environment.yml:25]: We should check that if we have a version of GeoUtils installed already in editable mode, it does not get overwritten with this environment.

File: setup.py [hunk 4]
```diff
@@ -1,22 +1,52 @@
...
+      long_description_content_type="text/markdown",
+      url='https://github.com/GlacioHack/xdem',
       author='The GlacioHack Team',
+      author_email="john@doe.com",  # This is needed for PyPi unfortunately.
       license='BSD-3',
       packages=['xdem'],
       install_requires=['numpy', 'scipy', 'rasterio', 'geopandas',
-                        'pyproj', 'tqdm', 'geoutils @ https://github.com/GlacioHack/GeoUtils/tarball/master', 'scikit-gstat', 'scikit-image'],
+                        'pyproj', 'tqdm', 'scikit-gstat', 'scikit-image'],
       extras_require={'rioxarray': ['rioxarray'], 'richdem': ['richdem'], 'pdal': [
```
Comment (adehecq) [inline setup.py:46]: This makes me think. In the setup.py it's possible to mention optional dependencies. Is it possible to do that in the conda environment file too? Or should we have a "base_environement.yml" and "opt_environment.yml"?

**Review by adehecq — DISMISSED** [review]
Apart from some thoughts, this seems good!

**Comment by erikmannerfelt** [comment]
I unfortunately had to undo the hacky solution with the post-install script, because it had all kinds of side-effects! One included the proj-data not being installed correctly (probably because of a mix between pip and conda) so the CI testing failed constantly. This should also solve the question by @adehecq  if it overrides an already installed instance of geoutils (it shouldn't anymore)

File: setup.py (hunk 4 above)
Comment (erikmannerfelt) [reply setup.py:29]: It's unfortunately not possible.. One would indeed have to have an external yml or txt file for that.

File: environment.yml (hunk 3 above)
Comment (erikmannerfelt) [reply environment.yml:25]: This works now with the latest commits.

**Review by adehecq — APPROVED** [review]

**Pull request merged**